File: 프로그래머스/Lv.2/43165. 타겟 넘버/타겟 넘버.cpp
#include <string>
#include <vector>

using namespace std;

int targetNum{};
int result{};
int numbersSize{};
void DFS(vector<int>& numbers, int idx, int sum);
int solution(vector<int> numbers, int target) {
    int answer = 0;
    
    numbersSize = numbers.size();
    targetNum = target;
    DFS(numbers,0,0);
    
    return result;
}

void DFS(vector<int>& numbers, int idx, int sum)
{
    if(idx == numbersSize)
    {
        if(sum == targetNum)
            result+=1;
        return;
    }
    DFS(numbers,idx+1,sum+numbers[idx]);
    DFS(numbers,idx+1,sum-numbers[idx]);
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cs
using System;

public class Solution {
    int[] numbers;
    int target;
    int answer;
    public int solution(int[] numbers, int target) {
        this.numbers = numbers;
        this.target = target;
        this.answer = 0;
        DFS(0,0);
        return answer;
    }
    
    private void DFS(int idx, int result)
    {
        if(idx == numbers.Length)
        {
            if(result == target)
                answer+=1;
            return;
        }
        
        DFS(idx+1,result-numbers[idx]);
        DFS(idx+1,result+numbers[idx]);
    }
}